runtime/lua/vim/filetype.lua

```diff
@@ -2449,6 +2449,11 @@ local function match_pattern(name, path, tail, pat)
       return false
     end
   end
+
+  if not name then
```

Comment:
Maybe move this to the start of the function so it exits right away if name is nil.